csharp/ql/lib/semmle/code/csharp/exprs/Call.qll

```diff
@@ -290,7 +303,54 @@ private predicate isValidExplicitParamsType(Parameter p, Type t) {
  * ```
  */
 class MethodCall extends Call, QualifiableExpr, LateBindableExpr, @method_invocation_expr {
-  override Method getTarget() { expr_call(this, result) }
+  override Method getTarget() {
+    expr_call(this, result)
+    or
+    result = this.getACandidateTarget()
+  }
+
+  private Method getACandidateTarget() {
+    not exists(Method m | expr_call(this, m)) and
+    exists(string name, ValueOrRefType alternativeQualifierType |
+      invocation_member_name(this, name) and
+      alternativeQualifierType =
+        this.getQualifier()
+            .getType()
+            .(ValueOrRefType)
+            .getABaseType*()
+            .getAnAmbiguousAlternativeType*() and
```

Comment:
What happens, if we only look at "one" level of alternatives? Does that cover most cases?

Reply:
I've done this in https://github.com/github/codeql/pull/14891/commits/a4eb47face7830c24f3b8a4ff28e5665c351f545